### DISCUSSION TITLE
Baking world transform

### DISCUSSION THREAD

**Discussion by Samsy** [donmccurdy/glTF-Transform#942] [General]
Hello ! 

I was reading the doc a could not find a way to bake a world matrix into a geometry position buffer and normals directly, something similar as Three.js : 

geometry.applyMatrix4( worldMatrix ) 

Is there something similar using GLTF-transform ?

**Comment by donmccurdy** [comment]
Hi @Samsy! Probably what you want is either `transformMesh` or `transformPrimitive`:

```js
import { transformMesh, transformPrimitive } from '@gltf-transform/functions';
import { fromScaling } from 'gl-matrix/mat4';

const matrix = fromScaling([], [2, 2, 2]);

// (A) transform mesh
const mesh = document.getRoot().listMeshes();
transformMesh(mesh, matrix);

// (B) transform prim
const prim = mesh.listPrimitives()[0];
transformPrimitive(prim, matrix);
```

The 4x4 matrix is an array of numbers, as defined in [`gl-matrix/mat4`](https://glmatrix.net/docs/module-mat4.html).

https://gltf-transform.donmccurdy.com/functions.html#transformmesh
